arelle/plugin/validate/ESEF_2022/__init__.py

```diff
@@ -169,17 +169,17 @@ def modelXbrlLoadComplete(modelXbrl: ModelXbrl) -> None:
             modelXbrl.error("ESEF.RTS.Art.6.a.noInlineXbrlTags",
                             _("RTS on ESEF requires inline XBRL, no facts were reported."),
                             modelObject=modelXbrl)
-        if modelXbrl.fileSource.isArchive:
+        if modelXbrl.fileSource.isArchive and modelXbrl.fileSource.dir is not None:
             for filename in modelXbrl.fileSource.dir:
                 validateEntity(modelXbrl, filename, modelXbrl.fileSource)
         else:
-            validateEntity(modelXbrl, modelXbrl.fileSource.url, modelXbrl.fileSource)
+            validateEntity(modelXbrl, cast(str, modelXbrl.fileSource.url), modelXbrl.fileSource)
```

Comment:
Is this correct if `FileSource.select` is called with a `list`?

Reply:
Good catch! Updated